apm-agent-core/src/main/java/co/elastic/apm/agent/metrics/MetricRegistry.java

```diff
@@ -208,7 +210,7 @@ private MetricSet getOrCreateMetricSet(Labels labels) {
         if (metricSet != null) {
             return metricSet;
         }
-        if (activeMetricSets.size() < METRIC_SET_LIMIT) {
+        if (activeMetricSets.size() < coreConfiguration.getMetricSetLimit()) {
```

Comment:
This is a bit misleading as it means that the limit can be dynamically updated. However, since metricsets are typically registered only once per startup, modifying this at runtime will only take affect on not-yet registered metrics.
Please keep the limit as a final field initialized through construction, so that it's more obvious and predictive.